src/adapter/src/coord/peek.rs

```diff
@@ -580,7 +588,12 @@ impl<S: Append + 'static> crate::coord::Coordinator<S> {
                 }
             })
             .map(move |resp| match resp {
-                PeekResponse::Rows(rows) => PeekResponseUnary::Rows(finishing.finish(rows)),
+                PeekResponse::Rows(rows) => {
+                    match finishing.finish(rows, usize::cast_from(max_result_size)) {
```

Comment:
I kinda want this check to be before `memo.extend` instead! Assume a very common case of most rows are distinct. In that case the extend is the thing that'll oom us, not the conversion to unary. Can you count bytes there (and multiply by the row count)? I think that's strictly better.

Reply:
I agree I just moved it. The one consequence is that constants results less than or equal to 10,000 will not be included in this check. I think that's probably fine? It's unlikely that 10,000 rows or less will OOM, and if we think it will then we should make that number smaller.